test/Microsoft.AspNet.Mvc.FunctionalTests/TagHelpersTest.cs

```diff
@@ -58,6 +60,37 @@ public class TagHelpersTest
 #endif
         }
 
+        [Fact]
+        public async Task CanRenderViewsWithTagHelpersAndUnboundDynamicAttributes_Encoded()
+        {
+            // Arrange
+            var server = TestHelper.CreateServer(_app, SiteName, services =>
+            {
+                _configureServices(services);
+                services.AddTransient<IHtmlEncoder, TestHtmlEncoder>();
```

Comment:
Should be moving away from `TestHtmlEncoder` (it doesn't need to exist anymore). Use `CommonTestEncoder` instead.